### ISSUE TITLE
Better support for choice lists with many choices

### ISSUE THREAD

**Issue by taurr** [cargo-generate/cargo-generate#400]
Using a `choice` placeholder with a lot of options is currently not feasible, as the number of choices scroll off the screen in the interactive mode and makes it next to impossible to select anything.

We should implement e.g. paging to handle when there are more choices than can naturally fit in the terminal.

It would be super if it also was possible to search in the choice list.

**Comment by taurr** [comment]
As info and testing, try out:

```shell
cargo generate --git taurr/embedded-template-rs application
```

it's my template for a complete embedded application using ferrous tooling, still in the works.

**Comment by sassman** [comment]
I see. But instead of us tackling this issue I'd recommend we upstream that issue to the `dialoguer` crate maintainers, since I cannot imagine we are the only ones interested in a solution for this issue.

I found https://github.com/mitsuhiko/dialoguer/issues/118 that is slightly but none that is directly addressing this very issue.

**Comment by sassman** [comment]
upstream issue is opened

**Comment by deg0nz** [comment]
> It would be super if it also was possible to search in the choice list.

Searching would be possible via recently merged `fuzzy-search` feature, but it currently does not support paging. (Just FYI)

**Comment by taurr** [comment]
Just tried using newest dialoquer (via git dependency), enabling the "fuzzy-select" feature, then in `interactive.rs`:64 changing to use `FuzzySelect::with_theme` instead of `Select::with_theme`

Works like a charm 😄 

Now we're just waiting for a fix to mitsuhiko/dialoguer#125 and a release, then this issue can be closed (with the above corrections).

**Comment by deg0nz** [comment]
I looked into the `cargo-generate` code and was able to identify the error that actually happens here. Please see my last comment in mitsuhiko/dialoguer#125 🙂

**Comment by sassman** [comment]
upstream issue is now solved, `0.9.0` makes an explicit `.paged` call obsolete (in fact it was completely removed). Will provide an update for it.
